### PR TITLE
Concurrency: Implement `resume(returning:)` and `resume(throwing:)` for Unsafe*Continuation.

### PR DESCRIPTION
**Pull request by jckarter**

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci please test

File: stdlib/public/Concurrency/Task.cpp [hunk 1]
```diff
@@ -355,3 +355,49 @@ void swift::swift_task_run(AsyncTask *taskToRun) {
...
+
+namespace {
+  
+/// Structure that gets filled in when a task is suspended by `withUnsafeContinuation`.
+struct AsyncContinuationContext {
+  // These fields are unnecessary for resuming a continuation.
+  void *Unused1;
+  void *Unused2;
+  // Pointer to where to store an error result.
+  SwiftError **ErrorResult;
```
Comment (aschwaighofer) [inline stdlib/public/Concurrency/Task.cpp:367]: The error pointer is stored directly. This should be a `SwiftError *ErrorResult`.

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by DougGregor** [comment]
@swift-ci please test Linux

**Comment by DougGregor** [comment]
@swift-ci please test macOS

**Comment by compnerd** [comment]
The `dispatch/dispatch.h` inclusion seems to break the build on certain targets :-(.  Up until now, the standard library hasn't depended on dispatch, only the compiler.  We _could_ do another build of dispatch for Linux and Windows where dispatch is not part of the system if necessary.

**Comment by rjmccall** [comment]
> The `dispatch/dispatch.h` inclusion seems to break the build on certain targets :-(. Up until now, the standard library hasn't depended on dispatch, only the compiler. We _could_ do another build of dispatch for Linux and Windows where dispatch is not part of the system if necessary.

I think we're going to have to do that so that at least the C portions of Dispatch can be a dependency of _Concurrency, at least when we're not configured to use some alternate implementation.

**Comment by jckarter** [comment]
Sorry, I had meant to `#if __APPLE__` that part out, since the implementation here was just a fakeout for testing purposes. Now that @rjmccall has landed his task runtime implementation, I can remove the dispatch dependency here.

**Pull request merged**